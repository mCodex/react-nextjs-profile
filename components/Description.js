import React, { PureComponent } from 'react';

import { Header, Container } from 'semantic-ui-react';

import withData from '../hocs/DataProvider';

class Description extends PureComponent {
  render() {
    const { data } = this.props;
    return (
      <Container
        fluid
        text
      >
        <Header as="h2">Description</Header>
        <p>{data.description}</p>
      </Container>
    );
  }
}

export default withData(Description);
