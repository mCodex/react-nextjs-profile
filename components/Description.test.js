import React from 'react';
import renderer from 'react-test-renderer';

import Description from './Description';

describe('Description component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Description />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
