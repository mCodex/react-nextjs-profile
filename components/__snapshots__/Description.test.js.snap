// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component renders correctly 1`] = `
<div
  className="ui text fluid container"
>
  <h2
    className="ui header"
  >
    <span
      className={undefined}
      style={undefined}
    >
      description
    </span>
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et tortor vel quam fringilla placerat. Proin elementum felis metus. Nullam ut eleifend leo. Morbi consectetur risus vitae vulputate condimentum. Nam vitae diam sapien. Vestibulum sed elementum diam. Ut placerat quis augue sed tristique. Donec convallis sem erat, nec iaculis dolor sagittis a. Maecenas sagittis euismod orci sed dapibus.
  </p>
</div>
`;
